lib/onebox/engine/twitter_status_onebox.rb

```diff
@@ -40,65 +72,123 @@ def twitter_api_credentials_present?
       end
 
       def raw
-        @raw ||= client.status(match[:id]).to_hash if twitter_api_credentials_present?
-      end
-
-      def access(*keys)
-        keys.reduce(raw) do |memo, key|
-          next unless memo
-          memo[key] || memo[key.to_s]
+        if twitter_api_credentials_present?
+          @raw ||= client.status(match[:id]).to_hash
+        else
+          super
         end
       end
 
       def tweet
-        client.prettify_tweet(raw)&.strip
+        if twitter_api_credentials_present?
+          client.prettify_tweet(raw)&.strip
+        else
+          twitter_data[:description].gsub(/“(.+?)”/im) { $1 } if twitter_data[:description]
+        end
       end
 
       def timestamp
-        date = DateTime.strptime(access(:created_at), "%a %b %d %H:%M:%S %z %Y")
-        user_offset = access(:user, :utc_offset).to_i
-        offset = (user_offset >= 0 ? "+" : "-") + Time.at(user_offset.abs).gmtime.strftime("%H%M")
-        date.new_offset(offset).strftime("%-l:%M %p - %-d %b %Y")
+        if twitter_api_credentials_present? && raw[:data][:created_at]
+          date = DateTime.strptime(raw[:data][:created_at], "%Y-%m-%dT%H:%M:%S.%L%z")
+          date.strftime("%-l:%M %p - %-d %b %Y")
+        end
       end
 
       def title
-        access(:user, :name)
+        if twitter_api_credentials_present? && raw[:includes][:users][0][:name]
+          raw[:includes][:users][0][:name]
```

Comment:
Can we use `raw.dig` here and similar other places where we're accessing attributes a few levels down? It feels like it would be safer if for some reason that particular attribute or its parent was missing.